d3ploy/NOInst.py

```diff
@@ -0,0 +1,265 @@
+    )
+    
+    deployed=ts.Int(
+        doc="The number of facilities initially deployed to this institution. "+ 
+              "This should match the initial facilities in the initial facilities list",
+        tooltip="Initial facilities of this institution",
+        uilabel="Initial Facilities"
+    )
+
+    record=ts.Bool(
+        doc="Indicates whether or not the institution should record it's output to text " +
+              "file outputs. The output files match the name of the demand commodity of the " +
+              "institution.",
+        tooltip="Boolean to indicate whether or not to record output to text file.",
+        uilabel="Record to Text",
+        default=False
+    )
+
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.commodity_supply = {}
+        self.commodity_demand = {}
+        self.fac_supply = {}
+
+    
+    def enter_notify(self):
+        super().enter_notify()
+        lib.TIME_SERIES_LISTENERS[self.supply_commod].append(self.extract_supply)
+        lib.TIME_SERIES_LISTENERS[self.demand_commod].append(self.extract_demand)         
+
+
+    def tock(self):
+        """
+        This is the tock method for the institution. Here the institution determines the difference
+        in supply and demand and makes the the decision to deploy facilities or not.     
+        """
+        time = self.context.time
+        diff, supply, demand = self.calc_diff(time)
+        if  diff < 0:
+            proto = random.choice(self.prototypes)
+            #proto_v = self.fac_supply[proto]
+            prod_rate = self.commodity_supply[time] / self.deployed
+            number = np.ceil(-1*diff/prod_rate)
+            self.deployed += number
+            i = 0
+            while i < number:
+                self.context.schedule_build(self, proto)
+                i+=1
+        if self.record:
+            f = open(self.demand_commod+".txt", 'a')
+            f.write("Time " + str(time) + " Deployed " + str(self.deployed) + " supply " + str(self.commodity_supply[time]) + " demand " +str(self.commodity_demand[time]) + "\n")   
+            f.close()  
+
+    def calc_diff(self, time):
+        """
+        This function calculates the different in supply and demand for a given facility
+        Parameters
+        ----------        
+        time : int
+            This is the time step that the difference is being calculated for.
+        Returns
+        -------
+        diff : double
+            This is the difference between supply and demand at [time]
+        supply : double
+            The calculated supply of the supply commodity at [time].
+        demand : double
+            The calculated demand of the demand commodity at [time]
+        """
+        try:
+            supply = self.calc_methods[self.calc_method](self, self.commodity_supply)
+        except:
+            supply = self.calc_methods['ma'](self, self.commodity_supply)
```

Comment:
This should not be a try-except (which should be reserved for truly exceptional cases).  This can be done in one line with a get.

```python
supply = self.calc_methods.get(self.calc_method, self.calc_methods['ma'])(self, self.commodity_supply)
```

Also feel free to split it up onto a couple of lines, if it makes it more readable